### PR TITLE
Add string metadata to celldata

### LINKED ISSUE #222: add string metadata
Display name, boundary condition, construction and modifier should pretty much cover most of what we need to get started.
display_name of the construction/modifier and not a representation of the whole object.

### PR DESCRIPTION
**Pull request by devangcx**
This PR adds string metadata to the cell data of the Polydata objects in honeybee-vtk. The following metadata is added;

- Display Name
- Boundary Condition
- Construction Display Name
- Modifier Display Name

resolves #222

### CODE REVIEW

File: honeybee_vtk/to_vtk.py [hunk 1]
```diff
@@ -100,32 +100,87 @@ def convert_sensor_grid(
...
+        display_name = [polydata.display_name] * num_of_cells
+        boundary_condition = [polydata.boundary_condition] * num_of_cells
+        construction_display_name = [polydata.construction_display_name] * num_of_cells
+        modifier_display_name = [polydata.modifier_display_name] * num_of_cells
+
+        metadata = {
+            "Display Name": display_name,
+            "Boundary Condition": boundary_condition,
+            "Construction Display Name": construction_display_name,
+            "Modifier Display Name": modifier_display_name
```
Comment (mostaphaRoudsari) [inline honeybee_vtk/to_vtk.py:152]: As I commented on Basecamp I think we can simplify the keys and use name, boundary condition, construction and modifier.

**Review by mostaphaRoudsari — COMMENTED** [review]
Hi @devngc, the code is clean but I think we can implement it as part of the polydata class and also name it slightly better.

I think we should separate the process of getting metadata from assigning them to Polydata.

For instance right now you have:

```python
    polydata = convert_face_3d(shade.geometry)
    polydata = _add_metadata_to_cells(polydata, shade, 'Shade')
```

I think it should look like:

```python
    polydata = convert_face_3d(shade.geometry)
    meta_data = polydata._get_metadata(shade)  # get metadata from a honeybee object but don't assign it
    polydata.add_data(meta_data)
```

File: honeybee_vtk/to_vtk.py (hunk 1 above)
Comment (devngc) [reply honeybee_vtk/to_vtk.py]: Complied

**Comment by devngc** [comment]
@mostaphaRoudsari, I believe I complied with both of your comments.

**Comment by mostaphaRoudsari** [comment]
I won't be able to review until later today. Feel free to merge in the changes if you are in a rush.

File: honeybee_vtk/to_vtk.py [hunk 2]
```diff
@@ -91,41 +91,42 @@ def convert_sensor_grid(
...
 def convert_shade(shade: Shade) -> PolyData:
-    shade_data = convert_face_3d(shade.geometry)
-    shade_data.identifier = shade.identifier
-    shade_data.display_name = shade.display_name
-    shade_data.type = 'Shade'
-    return shade_data
+    polydata = convert_face_3d(shade.geometry)
+    metadata = polydata._get_metadata(shade)
+    for key, value in metadata.items():
+        polydata.add_data(value, key, data_range=[0, 0])
```
Comment (mostaphaRoudsari) [inline honeybee_vtk/to_vtk.py:107]: Should we add a method for `ad_data_from_dict`?

**Review by mostaphaRoudsari — APPROVED** [review]
LGTM! Thanks @devngc.

File: honeybee_vtk/to_vtk.py (hunk 2 above)
Comment (devngc) [reply honeybee_vtk/to_vtk.py]: I added it.

**Pull request merged**